book/bq.md

```diff
@@ -388,39 +388,89 @@ You follow all the same procedures as shown in the form or controller style abov
   drush_backend_batch_process();
 ```
 
-From <https://drupal.stackexchange.com/questions/225273/how-can-execute-batch-api-over-cron-jobs>: 
+From <https://drupal.stackexchange.com/questions/225273/how-can-execute-batch-api-over-cron-jobs>:
 
 ```php
-function module_name_drush_command() {
-  $items = array();
-  $items['migrate-batch-list'] = array(
-    'description' => 'Migrate import match',
-    'examples' => array(
-      'drush migrate-batch-list' => 'Migrate import match',
-    ),
-  );
-return $items;
+<?php
+namespace Drupal\example\Drush\Commands;
+use Drupal\Core\Batch\BatchBuilder;
+use Drupal\Core\Database\Connection;
+use Drush\Attributes as CLI;
+use Drush\Commands\DrushCommands;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+/**
+ * A Drush commandfile.
+ *
+ * In addition to this file, you need a drush.services.yml
+ * in root of your module, and a composer.json file that provides the name
+ * of the services file to use.
+ */
+final class ExampleCommands extends DrushCommands {
+  /**
+   * Constructs an ExampleCommands object.
+   */
+  public function __construct(
+    private readonly Connection $connection,
+  ) {
+    parent::__construct();
+  }
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('database'),
+    );
+  }
+  /**
+   * Command description here.
+   */
+  #[CLI\Command(name: 'example:batch-import', aliases: ['ebi'])]
+  #[CLI\Usage(name: 'example:batch-import', description: 'Import data from provider with batch.')]
+  public function runBatchImportData() {
+    $provider_data = [
+      [
+        'id' => 1,
+        'body' => 'Lorem Ipsum',
+      ],
+      [
+        'id' => 2,
+        'body' => 'Lorem Ipsum 2',
+      ],
+    ];
+    $batch = new BatchBuilder();
+    foreach ($provider_data as $provider_datum) {
+      //You can use $this->connection to run a DB query to check if $provider_datum['id'] exists already in the database.
+      $batch->addOperation([ExampleCommands::class, 'batchImportData'], [$provider_datum]);
+    }
+    // We have set all operations in the batch.
+    // Now we have to add more info for batch.
+    $batch->setTitle('Import data from provider');
+    $batch->setFinishCallback([ExampleCommands::class, 'batchFinishImportData']);
+    batch_set($batch->toArray());
+    $this->logger()->notice("Batch is set.");
+    $this->logger()->notice("Start the batch process.");
+    drush_backend_batch_process();
+    return DrushCommands::EXIT_SUCCESS;
+  }
+  public static function batchImportData($data) {
+    // The structure of $data is not valid for $node_manager->create($data), it's only for showcase.
+    $node_manager = \Drupal::entityTypeManager()->getStorage('node');
```

Comment:
@dimitriskr I like this example.  Do you think we should tweak it a little to make it create some article nodes (for example) ?

Reply:
Sure. Any particular structure with fields or a general one?

Reply:
Great.  How about just fill in a title, body and author in an article node?  I just try to always have actual usable examples in the code if possible,

Reply:
@dimitriskr I hope I didn't scare you off with my suggestion

Reply:
@selwynpolit Haha no no, life happens. I'll get to it soon

Reply:
Is it better now? @selwynpolit 

(better late than never)

Reply:
looks great @dimitriskr Thanks!